src/Umbraco.Web/Editors/LanguageController.cs

```diff
@@ -56,6 +59,13 @@ public void DeleteLanguage(int id)
             {
                 throw new EntityNotFoundException(id, $"Could not find language by id: '{id}'.");
             }
+
+            if (language.IsDefaultVariantLanguage)
+            {
+                var message = $"Language with id '{id}' is currently set to 'default' and can not be deleted.";
```

Comment:
Although this will work, it is better to return validation messages so that the UI can show them, this will end up as a 500 error which means a ysod will show. I'll fix up and show you.

Reply:
ahh, sorry it will be the right error code but it won't bubble up the message correctly.